src/ape_test/_cli.py

```diff
@@ -12,9 +80,57 @@
     context_settings=dict(ignore_unknown_options=True),
 )
 @ape_cli_context()
+@click.option(
+    "-w",
+    "--watch",
+    is_flag=True,
+    default=False,
+    help="Watch for changes to project files and re-run the test suite with the given options.",
+)
+@click.option(
+    "--watch-folders",
+    multiple=True,
+    type=Path,
+    default=[Path("contracts"), Path("tests")],
+    help=(
+        "Folders to watch for changes using `ape test --watch`."
+        " Defaults to `contracts/` and `tests/`"
+    ),
+)
+@click.option(
```

Comment:
We should add the CLI configuration + the implementation code to the `pytest` integration module(s). Otherwise, there is a discrepancy between `pytest` and `ape test` all of a sudden. Currently, we are able to run either `pytest` or `ape test`. Otherwise `pytest` will only half-work, which causes confusion. 

History: shortly after we released 0.1.0, this confused one user in particular and we fixed it by allowing `pytest` to work fully with the `ape_test` integration when running `pytest` as the command. I was looking for the message or PR but couldn't find either though.

Additionally, it will allow us to test this using `pytester`, but that is really an aside.

Reply:
My main concern is that now `pytest` and `ape test` are different when before they weren't, but I suppose if you are okay with that then I won't make too big of a fuss bout it.

Reply:
Yeah, I wouldn't want this to work with `pytest` because I would use `pytest-watch` for this functionality instead